host/tests/dsCompositeIn_L3_Tests/dsCompositeIn_test1_VerifyConnect_Callback.py

```diff
@@ -0,0 +1,123 @@
+# *
+#* ******************************************************************************
+
+import os
+import sys
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(dir_path, "../"))
+
+from dsCompositeInHelperClass import dsCompositeInHelperClass
+
+class dsCompositeIn_test1_VerifyConnect_Callback(dsCompositeInHelperClass):
+    """
+    Test class to connect, disconnect and verify the status of CompositeIn device
+
+    This class uses the `dscompositeInClass` to interact with the device's compositeIn ports,
+    downloading necessary test assets, selecting
+    CompositeIn ports, and performing verification of compositeIn output.
+    """
+    def __init__(self):
+        """
+        Initializes the test1_VerifyConnect_Callback test .
+
+        Args:
+            None.
+        """
+        self.testName  = "test1_VerifyConnect_Callback"
+        super().__init__(self.testName, '1')
+
+    #TODO: Current version supports only manual verification.
+    def testPlugUnplugtCompositeIn(self, Connect:False, manual=False, port_type:str=0):
+        """
+        Verifies whether the compositeIn connected or not.
+
+        Args:
+            disconnect (bool): Connect or Disconnect compositeIn source device
+            manual (bool, optional): Manual verification (True: manual, False: other verification methods).
+                                     Defaults to other verification methods
+
+        Returns:
+            bool
+        """
+        if manual == True:
+            if Connect == True:
+                return self.testUserResponse.getUserYN(f"Connect the compositeIn source device to Port {port_type} and press Y:")
+
+            else :
+                return self.testUserResponse.getUserYN(f"Disconnect the compositeIn source device to Port {port_type} and press Y:")
+        else :
+            #TODO: Add automation verification methods
+            return False
+
+    def testFunction(self):
+        """This function will verify the connect/ disconnect status of CompositeIn device.
+
+        This function:
+        - Connect / disconnect the compositeIn port.
+        - Verify the status through callbacks
+
+        Returns:
+            bool: Final result of the test.
+        """
+
+        self.log.testStart("test1_VerifyConnect_Callback", '1')
+
+        # Initialize the dsCompositeIn module
+        self.testdsCompositeIn.initialise()
+
+        # Loop through the supported ports
+        for port in self.testdsCompositeIn.getSupportedPorts():
+            self.log.stepStart(f'Connect Status Test for Port {port}')
+
+            portstr = f"dsCOMPOSITE_IN_PORT_{port}"
+            result = self.testPlugUnplugtCompositeIn(True,True, portstr)
+
+            status = self.testdsCompositeIn.getConnectionCallbackStatus()
+            if status != None and portstr == status[0] and status[1] == True:
+                result = True
+            else:
+                result = False
```

Comment:
self.log.stepResult(
    status is not None and portstr == status[0] and status[1] == True,
    f'Connect Status Test for Port {portstr}'
)

can we try something like this?

Reply:
done